carbon-design-system/carbon
# fix(combobox): controlled combobox clear

Closes https://github.com/carbon-design-system/carbon/issues/18027

* update the way onChange is called in comboBox to prevent duplicate calls
* add unit test cases for controlled combobox managed by parent component
* fix console errors related to missing onChange prop from ComboBox.mdx
* fix console errors for components using WithLayer related to wrong proptypes
![Screenshot 2024-11-14 at 11 15 19](https://github.com/user-attachments/assets/d9d714d0-0980-45d1-adce-7c836560b208)

#### Testing / Reviewing

- add log to [onChange in fullyControlled combobox in ComboBox.stories.js](https://github.com/Neues/carbon/blob/ef9f8b0418431d885f464effeff4ab98bbea78e8/packages/react/src/components/ComboBox/ComboBox.stories.js#L257)
- select different options to see onChange fires only once
- select the same option to see that onChange does not fire
- add log to onChange other ComboBox stories to ensure it is firing the appropriate amount of times
- verify console errors are no longer present
